### PR TITLE
Add more unit tests by GitHub Copilot Chat

### PR DESCRIPTION
**Pull request by louislam**
Always think writing unit tests is too boring, hope GitHub Copilot Chat could help me to do the job.

image

### CODE REVIEW

**Comment by Zaid-maker** [comment]
Dude is using light theme Omg! 😳

**Comment by louislam** [comment]
> Dude is using light theme Omg! 😳

It is not my usual environment. I don't use VSCode (GitHub Codespace), but GitHub Copilot Chat is not available for JetBrains' IDE.

But anyway, I am still using dark theme for UI and light theme for code with slightly grey background. Plus, I am also using a non-fixed width font. Every developers are going to hate me now 🤣, but I love it.

image

**Comment by Zaid-maker** [comment]
> > Dude is using light theme Omg! 😳
> 
> It is not my usual environment. I don't use VSCode (GitHub Codespace), but GitHub Copilot Chat is not available for JetBrains' IDE.
> 
> But anyway, I am still using dark theme for UI and light theme for code with slightly grey background. Plus, I am also using a non-fixed width font. Every developers are going to hate me now 🤣, but I love it.

Bless your eyes DUDE!

File: test/cypress/unit/util-frontend.spec.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
-import { hostNameRegexPattern } from "../../../src/util-frontend";
+import { hostNameRegexPattern,  } from "../../../src/util-frontend";
```
Comment (Saibamen) [inline test/cypress/unit/util-frontend.spec.js:1]:
```suggestion
import { hostNameRegexPattern } from "../../../src/util-frontend";
```

File: test/cypress/unit/util-frontend.spec.js [hunk 2]
```diff
@@ -39,3 +40,19 @@ describe("Test util-frontend.js", () => {
...
+
+describe('Test ucfirst', () => {
+    it('should capitalize the first letter of a string', () => {
+        expect(ucfirst('hello')).toBe('Hello');
+        expect(ucfirst('world')).toBe('World');
+    });
+
+    it('should return an empty string if the input is empty', () => {
+        expect(ucfirst('')).toBe('');
+    });
```
Comment (Saibamen) [inline test/cypress/unit/util-frontend.spec.js:52]: Do this same for null and/or undefined

**Comment by louislam** [comment]
Close as it is not as smart as I think.

**Pull request closed without merging**